tests/test_protocols/test_generator/test_common.py

```diff
@@ -0,0 +1,371 @@
+
+        content_type_1 = "FrozenSet[int]"
+        assert (
+            _union_sub_type_to_protobuf_variable_name(content_name, content_type_1)
+            == "proposal_type_set_of_int"
+        )
+
+        content_type_2 = "Tuple[str, ...]"
+        assert (
+            _union_sub_type_to_protobuf_variable_name(content_name, content_type_2)
+            == "proposal_type_list_of_str"
+        )
+
+        content_type_3 = "Dict[bool, float]"
+        assert (
+            _union_sub_type_to_protobuf_variable_name(content_name, content_type_3)
+            == "proposal_type_dict_of_bool_float"
+        )
+
+        content_type_4 = "int"
+        assert (
+            _union_sub_type_to_protobuf_variable_name(content_name, content_type_4)
+            == "proposal_type_int"
+        )
+
+        content_type_5 = "DataModel"
+        assert (
+            _union_sub_type_to_protobuf_variable_name(content_name, content_type_5)
+            == "proposal_type_DataModel"
+        )
+
+    def test_python_pt_or_ct_type_to_proto_type(self,):
+        """Test the '_python_pt_or_ct_type_to_proto_type' method"""
+        content_type_bytes = "bytes"
+        assert _python_pt_or_ct_type_to_proto_type(content_type_bytes) == "bytes"
+
+        content_type_int = "int"
+        assert _python_pt_or_ct_type_to_proto_type(content_type_int) == "int32"
+
+        content_type_float = "float"
+        assert _python_pt_or_ct_type_to_proto_type(content_type_float) == "float"
+
+        content_type_bool = "bool"
+        assert _python_pt_or_ct_type_to_proto_type(content_type_bool) == "bool"
+
+        content_type_str = "str"
+        assert _python_pt_or_ct_type_to_proto_type(content_type_str) == "string"
+
+        content_type_ct = "Query"
+        assert _python_pt_or_ct_type_to_proto_type(content_type_ct) == "Query"
+
+    def test_includes_custom_type(self,):
+        """Test the '_includes_custom_type' method"""
+        content_type_includes_1 = "Optional[DataModel]"
+        assert _includes_custom_type(content_type_includes_1) is True
+
+        content_type_includes_2 = "Union[int, DataModel]"
+        assert _includes_custom_type(content_type_includes_2) is True
+
+        content_type_includes_3 = "Optional[Union[int, float, DataModel, Query, float]]"
+        assert _includes_custom_type(content_type_includes_3) is True
+
+        content_type_not_includes_1 = "Optional[int]"
+        assert _includes_custom_type(content_type_not_includes_1) is False
+
+        content_type_not_includes_2 = "Union[int, float, str]"
+        assert _includes_custom_type(content_type_not_includes_2) is False
+
+        content_type_not_includes_3 = (
+            "Optional[Union[int, float, FrozenSet[int], Tuple[bool, ...], float]]"
+        )
+        assert _includes_custom_type(content_type_not_includes_3) is False
+
+    def test_is_installed(self,):
+        """Test the 'is_installed' method"""
+        # ToDo
+        pass
```

Comment:
It;s enough to call the method and then mock the response.

Reply:
Done